Sources/CoreCommands/SwiftCommandState.swift

```diff
@@ -1064,25 +1064,33 @@ public final class SwiftCommandState {
         // Try a non-blocking lock first so that we can inform the user about an already running SwiftPM.
         do {
             try workspaceLock.lock(type: .exclusive, blocking: false)
+            let pid = ProcessInfo.processInfo.processIdentifier
+            try String(pid).write(toFile: self.scratchDirectory.appending(".lock").pathString, atomically: true, encoding: .utf8)
         } catch ProcessLockError.unableToAquireLock(let errno) {
             if errno == EWOULDBLOCK {
+                let lockingPID = try? String(contentsOfFile: self.scratchDirectory.appending(".lock").pathString, encoding: .utf8)
+                let pidInfo = lockingPID.map { "(PID: \($0))" } ?? ""
```

Comment:
I'd add a space after the closing parenthesis, like so:

```swift
let pidInfo = lockingPID.map { "(PID: \($0)) " } ?? ""
```

...and then correspondingly remove the space after the `pifInfo` interpolation a few lines down, like so:

```swift
"Another instance of SwiftPM \(pidInfo)is already running using '\(self.scratchDirectory)', but this will be ignored since `--ignore-lock` has been passed"
```

...so that we don't end up with a double space between "SwiftPM" and "is" if we couldn't read the PID, since that would look out of place.